src/bundle/Resources/public/js/scripts/fieldType/ezobjectrelationlist.js

```diff
@@ -0,0 +1,212 @@
+    const SELECTOR_BTN_ADD = '.ez-relations__table-action--create';
+    const SELECTOR_ROW = '.ez-relations__item';
+    const EVENT_CUSTOM = 'validateInput';
+
+    class EzObjectRelationListValidator extends global.eZ.BaseFieldValidator {
+        /**
+         * Validates the input
+         *
+         * @method validateInput
+         * @param {Event} event
+         * @returns {Object}
+         * @memberof EzObjectRelationListValidator
+         */
+        validateInput({currentTarget}) {
+            const isRequired = currentTarget.required;
+            const isEmpty = !currentTarget.value.length;
+            const hasCorrectValues = currentTarget.value.split(',').every(id => !isNaN(parseInt(id, 10)));
+            const label = currentTarget.closest(SELECTOR_FIELD).querySelector('.ez-field-edit__label').innerHTML;
+            const result = { isError: false };
+
+            if (isRequired && isEmpty) {
+                result.isError = true;
+                result.errorMessage = global.eZ.errors.emptyField.replace('{fieldName}', label);
+            } else if (!isEmpty && !hasCorrectValues) {
+                result.isError = true;
+                result.errorMessage = global.eZ.errors.invalidValue.replace('{fieldName}', label);
+            }
+
+            return result;
+        }
+    }
+
+    [...document.querySelectorAll(SELECTOR_FIELD)].forEach(fieldContainer => {
+        const validator = new EzObjectRelationListValidator({
+            classInvalid: 'is-invalid',
+            fieldContainer,
+            eventsMap: [
+                {
+                    selector: SELECTOR_FIELD_INPUT,
+                    eventName: 'blur',
+                    callback: 'validateInput',
+                    errorNodeSelectors: [SELECTOR_LABEL_WRAPPER]
+                },
+                {
+                    isValueValidator: false,
+                    selector: SELECTOR_FIELD_INPUT,
+                    eventName: EVENT_CUSTOM,
+                    callback: 'validateInput',
+                    errorNodeSelectors: [SELECTOR_LABEL_WRAPPER]
+                }
+            ]
+        });
+        const udwContainer = document.getElementById('react-udw');
+        const token = document.querySelector('meta[name="CSRF-Token"]').content;
+        const siteaccess = document.querySelector('meta[name="SiteAccess"]').content;
+        const sourceInput = fieldContainer.querySelector(SELECTOR_INPUT);
+        const relationsContainer = fieldContainer.querySelector('.ez-relations__list');
+        const relationsWrapper = fieldContainer.querySelector('.ez-relations__wrapper');
+        const relationsCTA = fieldContainer.querySelector('.ez-relations__cta');
+        const addBtn = fieldContainer.querySelector(SELECTOR_BTN_ADD);
+        const selectedItemsLimit = parseInt(relationsContainer.dataset.limit, 10);
+        const closeUDW = () => udwContainer.innerHTML = '';
+        const onCancel = () => closeUDW();
```

Comment:
Do we need this? Couldn't you just set in UDW: `onCancel: closeUDW`?

Reply:
That's another option. Good point! :)